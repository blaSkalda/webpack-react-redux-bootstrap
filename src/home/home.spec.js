/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from './home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {
  const props = {
    messages: {
      data: [],
      fetching: false,
    },
    getMessages: jest.fn(),
    getMessage: jest.fn(),
  };

  it('renders <Home />', () => {
    const wrapper = shallow(<Home {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders <Home /> component and triggers get messages', () => {
    const wrapper = shallow(<Home {...props} />);
    props.getMessages.mockReset();

    wrapper.find('#get-messages').simulate('click');

    expect(props.getMessages.mock.calls.length).toEqual(1);
  });

  it('renders <Home /> component and triggers get message', () => {
    const wrapper = shallow(<Home {...props} />);
    props.getMessage.mockReset();

    wrapper.find('#get-message').simulate('click');

    expect(props.getMessage.mock.calls.length).toEqual(1);
  });
});
