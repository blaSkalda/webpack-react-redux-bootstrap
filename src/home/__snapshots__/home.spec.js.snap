// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders <Home /> 1`] = `
<div
  className="home-page"
>
  <header>
    <div
      className="navbar navbar-dark bg-dark shadow-sm"
    >
      <div
        className="container d-flex justify-content-between"
      >
        <a
          className="navbar-brand d-flex align-items-center"
          href="#home"
        >
          <strong>
            React/Redux quick start
          </strong>
        </a>
      </div>
    </div>
  </header>
  <div
    className="album py-5 bg-light"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <button
            className="btn btn-primary"
            id="get-messages"
            onClick={[Function]}
            type="button"
          >
            Get messages
          </button>
        </div>
        <div
          className="col-sm"
        >
          <button
            className="btn btn-secondary"
            id="get-message"
            onClick={[Function]}
            type="button"
          >
            Get message 1
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        />
      </div>
    </div>
  </div>
</div>
`;
